storage-proofs-porep/src/stacked/vanilla/proof.rs

```diff
@@ -103,9 +103,12 @@ impl<'a, Tree: 'static + MerkleTreeTrait, G: 'static + Hasher> StackedDrg<'a, Tr
             let mut parents = vec![0; base_degree];
             graph.base_parents(x, &mut parents)?;
 
-            for parent in &parents {
-                columns.push(t_aux.column(*parent)?);
-            }
+            columns.extend(
+                parents
+                    .into_par_iter()
+                    .map(|parent| t_aux.column(parent).expect("failed to get parent column"))
+                    .collect::<Vec<Column<Tree::Hasher>>>(),
```

Comment:
you should be able to collect into `Result<Vec>>` allowing you to acutally handle the error, instead of `expect`ing

Reply:
Maybe I'm missing something, but it looks like not what I want.  The return type is `Result<Vec<Column<Tree::Hasher>>>`, but w/o the `expect`, we'd be returning `Vec<Result<Column<Tree::Hasher>>>`.

Reply:
```rust
parents
  .into_par_iter()
  .map(|parent| t_aux.column(parent))
  .collect::<Result<Vec<Column<Tree::Hasher>>, _>>()?
```

is what I have in mind, to avoid  the `expect`

Reply:
Hrm, that makes sense, I think I had simply noted the difference of return type if the expect was removed, not changing the collect type.  Is there a difference between:

```
.collect::<Result<Vec<Column<Tree::Hasher>>, _>>()?
```

and

```
.collect::<Result<Vec<Column<Tree::Hasher>>>()?
```

They both appear to work.

Reply:
> They both appear to work.

Nevermind, just seems to be the type of error, which is implied either way (explicitly or implicitly).